corehq/apps/users/models.py

```diff
@@ -1334,6 +1334,13 @@ def set_location(self, location):
         Set the location, and all important user data, for
         the user.
         """
+        if not location:
+            # Remove associated info
+            self.user_data.pop('commtrack-supply-point', None)
+            self.user_data.pop('commcare_primary_case_sharing_id', None)
+            self.location_id = None
+            return
+
```

Comment:
I think you might need to add a check that is something like:
```
if self.project.supports_multiple_locations_per_user:
    self.clear_locations()
```

though it also looks like there might be some problems with that function